apps/api/v1/pages/api/event-types/_post.ts

```diff
@@ -275,24 +275,58 @@ async function postHandler(req: NextApiRequest) {
     children: _,
     ...parsedBody
   } = schemaEventTypeCreateBodyParams.parse(body || {});
+  if (parsedBody.teamId && parsedBody.parentId) {
+    //parentId and teamId can't be used together as parentId is present only for child event types of a managed event
+    //teamId is used for event types belonging to a team
+    //child events of a team event do not have teamId , they are identified via parentId
+    //throw error to avoid unexpected behaviour
+    throw new HttpError({
+      statusCode: 400,
+      message: "`parentId` and `teamId` both cannot be present in the request",
+    });
+  }
+  if (parsedBody.teamId && parsedBody.userId) {
+    //team event is not associated with an owner or user
+    throw new HttpError({
+      statusCode: 400,
+      message: "`teamId` and `userId` both cannot be present in the request",
+    });
+  }
+  if (!parsedBody.teamId && parsedBody.schedulingType) {
+    //schedulingType is applicable only for team events
+    throw new HttpError({
+      statusCode: 400,
+      message: "schedulingType is applicable only for team events",
+    });
+  }
 
   let data: Prisma.EventTypeCreateArgs["data"] = {
     ...parsedBody,
-    userId,
-    users: { connect: { id: userId } },
     bookingLimits: bookingLimits === null ? Prisma.DbNull : bookingLimits,
     durationLimits: durationLimits === null ? Prisma.DbNull : durationLimits,
   };
 
+  if (!parsedBody.teamId) {
+    //connect user if eventtype doesn't belong to a team
+    data = {
+      ...data,
+      userId,
+      users: { connect: { id: userId } },
+    };
+  }
+
   await checkPermissions(req);
 
+  //user with admin or owner role on team can create child event types
+  let isChildEventTypeForTeamWithAccess = false;
```

Comment:
NIT: Is `isChildEventTypeForTeamWithAccess` necessary? If user does not have permissions errors are being thrown inside the functions. You can use paresedBody.parentId for if condition instead of this variable

Reply:
parsedBody.parentId may confuse someone revisiting the code after a long time.
I thought of using a variable just to make it easier to go through code while revisiting in future

Let me know your thoughts ,  if we should prefer not having a variable here.

Reply:
I think a comment above the if statement should suffice.

Reply:
Updated PTAL